osu.Server.Spectator/Hubs/ServerMultiplayerRoom.cs

```diff
@@ -103,48 +103,24 @@ private MatchTypeImplementation createTypeImplementation(MatchType type)
         /// </summary>
         /// <param name="countdown">The countdown to start. The <see cref="MultiplayerRoom"/> will receive this object for the duration of the countdown.</param>
         /// <param name="onComplete">A callback to be invoked when the countdown completes.</param>
-        public void StartCountdown(MultiplayerCountdown countdown, Func<ServerMultiplayerRoom, Task> onComplete)
+        public async Task StartCountdown(MultiplayerCountdown countdown, Func<ServerMultiplayerRoom, Task> onComplete)
         {
-            StopCountdown();
+            await StopCountdown();
 
             var stopSource = countdownStopSource = new CancellationTokenSource();
             var skipSource = countdownSkipSource = new CancellationTokenSource();
 
-            Task lastCountdownTask = countdownTask;
-            countdownTask = start();
+            countdownStartTime = DateTimeOffset.Now;
+            countdownDuration = countdown.TimeRemaining;
 
-            // Some sections in the following code require single-threaded execution mode. This is achieved by re-retrieving the room from the hub, forcing an exclusive lock on it.
-            async Task start()
-            {
-                // Wait for the last countdown to finalise before starting a new one.
-                try
-                {
-                    await lastCountdownTask;
-                }
-                catch
-                {
-                    // Any failures in the last countdown should not prevent future countdowns from running.
-                }
+            Countdown = countdown;
 
-                // Notify users that a new countdown has started.
-                // Note: The room must be re-retrieved rather than using our own instance to enforce single-thread access.
-                using (var roomUsage = await hub.GetRoom(RoomID))
-                {
-                    if (roomUsage.Item == null)
-                        return;
-
-                    // The countdown could have been cancelled in a separate request before this task was able to run.
-                    if (stopSource.IsCancellationRequested)
-                        return;
-
-                    roomUsage.Item.Countdown = countdown;
-
-                    countdownStartTime = DateTimeOffset.Now;
-                    countdownDuration = countdown.TimeRemaining;
+            countdownTask = start();
 
-                    await hub.NotifyNewMatchEvent(roomUsage.Item, new CountdownChangedEvent { Countdown = countdown });
-                }
+            await hub.NotifyNewMatchEvent(this, new CountdownChangedEvent { Countdown = countdown });
```

Comment:
I'd prefer this to be sent before the task is started to avoid potential misordering of events.